archie-utils/src/main/java/com/nedap/archie/json/ExcludeEmptyCollectionsFilter.java

```diff
@@ -0,0 +1,30 @@
+package com.nedap.archie.json;
+
+import java.util.Collection;
+import java.util.Map;
+
+public class ExcludeEmptyCollectionsFilter {
+    @Override
+    // Return true to exclude
+    public boolean equals(Object o) {
+
+        // Exclude Null
+        if (o == null) {
+            return true;
+        }
+
+        if (o instanceof Map) {
+            return ((Map) o).size() == 0;
+        }
+        if (o instanceof Collection) {
+            return ((Collection) o).size() == 0;
+        }
+
+        if (o.getClass().isArray()) {
+            return ((Object[]) o).length == 0;
+        }
```

Comment:
This crashes on DvMultimedia objects with empty data arrays:
```java
        DvMultimedia dvMultimedia = new DvMultimedia();
        dvMultimedia.setData(new byte[0]);
```
```
com.fasterxml.jackson.databind.JsonMappingException: class [B cannot be cast to class [Ljava.lang.Object; ([B and [Ljava.lang.Object; are in module java.base of loader 'bootstrap') (through reference chain: com.nedap.archie.rm.datavalues.encapsulated.DvMultimedia["data"])
```

Reply:
I think it is fine if an empty byte[] is serialized as [] anyway. So, I am changing it to only detect Object[]